greeter/README.md

```diff
@@ -1,21 +1,474 @@
-- Create a fr greeter
-- Register it to the app
-- Try endpoint again
+Rewritten in TypeScript from ground-up, LoopBack 4 core offers an Inversion of Control container with powerful Dependency Injection capabilities as the foundation to build large-scale Node.js projects with great flexibility and extensibility. LoopBack also integrates with cloud native health, metrics, and tracing to facilitate development of Microservices for cloud deployment.
 
-# Step 3
+## About This Workshop
 
-- Add caching observer
-- Add caching interceptor
-- Add caching service
\ No newline at end of file
+As your Node.js application becomes more complex, how do you increase its scalability with proven design patterns and best practices? How do you follow 12-factor principles to make your application flexible, extensible and composable? Let’s run a fun workshop to create a scalable and extensible application! While we could do this from scratch, we'll get there much faster and easier using a framework that does a lot of the hard work for us.
+
+In this workshop, you will be guided to scaffold a greeter app using the LoopBack 4 framework, and create a simple greeting API that prints out a welcome message, then enrich it without modifying its implementation in the next 4 steps:
+
+1. The greeting service is packaged as a separate module from the greeting app (modular)
+2. The greeting API can speak different languages based on the client locale. The support for new languages can be plugged in (extensibility and pluggability)
+3. Enable the caching with sweeping for REST requests (interception/observation)
+4. The greeting app and certain greeters have configuration options (configurability) _FIXME: we don't have this part, right?_
+
+![Greeting application diagram](https://raw.githubusercontent.com/strongloop/loopback-next/master/examples/greeting-app/greeting-app.png)
+
+This miniature project reflects many key aspects of a scalable and extensible applications, and relevant techniques will be covered in the workshop, like Inversion of Control, Dependency Injection, Decorators and Component.
+
+
+## Before We Begin
+To complete the steps in this tutorial, you need to install Node.js and the LoopBack 4 command line interface (CLI).
+
+1. Install Node.js version 8.9 or higher. 
+    See the [Node documentation](https://nodejs.org/en/download/) for installation instructions.
+    
+    Tip: We recommend to use [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) to make it easier to manage the Node.js versions you're using. 
+
+2. Install LoopBack 4 CLI
+
+    The LoopBack 4 CLI is a command-line interface that scaffolds an application as well as artifacts (for example, controllers, models, and repositories) for existing applications. The CLI provides the fastest and easiest way to get started with a LoopBack 4 project that adheres to best practices.
+
+    Install the CLI globally by running the following command:
+
+    ```sh
+    npm i -g @loopback/cli
+    ```
```

Comment:
Shall we recommend a code editor here?
Like 

3. Recommended code editor

Choose a code editor you like for incrementally adding features in the workshop. We recommend [Visual Studio Code](https://code.visualstudio.com/), which provides an awesome integration for TypeScript projects and built-in features like auto import to improve the dev experience. A LoopBack app created from cli will include vscode setting files.

Reply:
Thanks.  I'd prefer to use the term "development environment" we used in https://github.com/strongloop/loopback-next/blob/master/docs/site/DEVELOPING.md